components/chart/header/index.tsx

```diff
@@ -1,10 +1,80 @@
+import 'react-tagsinput/react-tagsinput.css';
+import React, { useState } from 'react';
+import BCTagsInput from './bctags-input';
+import Inputtext from './inputText';
+
 export default function Header({
   showModulesSearch,
   chartName,
+  predefinedTags,
 }: {
   showModulesSearch: (state: boolean) => void;
   chartName: string;
+  predefinedTags: string[];
 }) {
+  const [tags, setTags] = useState([]);
+  const [selectedOption, setSelectedOption] = useState(null);
+  predefinedTags = [
+    'العقيدة',
+    'الفقه',
+    'التفسير',
+    'الحديث',
+    'القراءات',
+    'أصول التفسير',
+    'أصول الفقه',
+    'النحو والصرف',
+    'البلاغة',
+    'المنطق',
+    'مصطلح الحديث',
+    'القواعد الفقهية',
+    'التاريخ',
+    'التراجم',
+    'أصول الدين'
+  ];
+  const handleTagsChange = (newTags: string[]) => {
+    setTags(newTags);
+  };
+
+  const handleSelectorChange = (selectedOption) => {
+    if (selectedOption && !tags.includes(selectedOption)) {
+      setTags([selectedOption,...tags]);
+      handleCopyURL();
```

Comment:
No need to copy the URL, but we need to add the tag parameter to the URL, when I tested that didn't work